services/app-web/src/pages/MccrsId/MccrsId.tsx

```diff
@@ -0,0 +1,181 @@
+
+    const mccrsIDInitialValues: MccrsIdFormValues = {
+        mccrsId: undefined,
+    }
+
+    const showFieldErrors = (error?: FormError) =>
+        shouldValidate && Boolean(error)
+
+    const handleFormSubmit = async (values: MccrsIdFormValues) => {
+        setShowPageErrorMessage(false)
+        try {
+            const updateResult = await updateFormData({
+                variables: {
+                    input: {
+                        mccrsID: values?.mccrsId?.toString(),
+                        id: id,
+                    },
+                },
+            })
+
+            const updatedSubmission: HealthPlanPackage | undefined =
+                updateResult?.data?.updateContract.pkg
+
+            if (!updatedSubmission) {
+                setShowPageErrorMessage(true)
+
+                console.info('Failed to update form data', updateResult)
+                recordJSException(
+                    `MCCRSIDForm: Apollo error reported. Error message: Failed to update form data ${updateResult}`
+                )
+                return new Error('Failed to update form data')
+            }
+            navigate(`/submissions/${updatedSubmission.id}`)
+        } catch (serverError) {
+            setShowPageErrorMessage(true)
+            recordJSException(
+                `MCCRSIDForm: Apollo error reported. Error message: ${serverError.message}`
+            )
+            return new Error(serverError)
+        }
+    }
+
+    return (
+        <>
+            {showPageErrorMessage && <GenericApiErrorBanner />}
+            <Formik
+                initialValues={mccrsIDInitialValues}
+                onSubmit={(values) => {
+                    return handleFormSubmit(values)
+                }}
+                validationSchema={() => MccrsIdFormSchema()}
+            >
+                {({ values, errors, handleSubmit, isSubmitting }) => (
+                    <>
+                        <UswdsForm
+                            className={styles.formContainer}
+                            id="MCCRSIDForm"
+                            aria-label="MCCRS ID Form"
+                            aria-describedby="form-guidance"
+                            onSubmit={(e) => {
+                                setShouldValidate(true)
+                                handleSubmit(e)
+                            }}
+                        >
+                            <fieldset className="usa-fieldset">
+                                <h3>MC-CRS record number</h3>
+                                <legend className="srOnly">
+                                    Add MC-CRS record number
+                                </legend>
+                                <FieldTextInput
+                                    name={'mccrsId'}
+                                    id={'mccrsId'}
+                                    label="Enter the Managed Care Contract and Rate Review System (MC-CRS) record number."
```

Comment:
We should trim the empty spaces before the MC-CRS number, but should we trim spaces in general? For example, `12   34`.

This is what the URL looks like with spaces before the ID
![Screenshot 2023-10-24 at 3 31 16 PM](https://github.com/Enterprise-CMCS/managed-care-review/assets/98117700/0c0d2124-ed95-4b6a-801d-4d6afe25230b)

![Screenshot 2023-10-24 at 3 31 31 PM](https://github.com/Enterprise-CMCS/managed-care-review/assets/98117700/086f094a-96a7-4525-a005-382c82093bec)